telegram/src/main/kotlin/com/github/kotlintelegrambot/network/serialization/GsonFactory.kt

```diff
@@ -0,0 +1,24 @@
+package com.github.kotlintelegrambot.network.serialization
+
+import com.github.kotlintelegrambot.entities.TelegramFile
+import com.github.kotlintelegrambot.entities.inlinequeryresults.InlineQueryResult
+import com.github.kotlintelegrambot.entities.inputmedia.GroupableMedia
+import com.github.kotlintelegrambot.network.serialization.adapter.GroupableMediaAdapter
+import com.github.kotlintelegrambot.network.serialization.adapter.InlineQueryResultAdapter
+import com.github.kotlintelegrambot.network.serialization.adapter.InputMediaAdapter
+import com.github.kotlintelegrambot.network.serialization.adapter.TelegramFileAdapter
+import com.google.gson.Gson
+import com.google.gson.GsonBuilder
+
+object GsonFactory {
+
+    fun createForApiClient(): Gson = GsonBuilder()
+        .registerTypeAdapter(InlineQueryResult::class.java, InlineQueryResultAdapter())
+        .create()
+
+    fun createForMultipartBodyFactory(): Gson = GsonBuilder()
+        .registerTypeAdapter(TelegramFile.ByFile::class.java, TelegramFileAdapter())
+        .registerTypeAdapter(TelegramFile::class.java, TelegramFileAdapter())
+        .registerTypeAdapter(GroupableMedia::class.java, GroupableMediaAdapter(InputMediaAdapter()))
+        .create()
```

Comment:
I've created different factory methods here to create Gson instances with different configurations to only have the adapters needed in each component and not having all of them in all the instances